dataflowAPI/src/RoseInsnFactory.C

```diff
@@ -442,7 +442,7 @@ void RoseInsnAMDGPUFactory::massageOperands(const Instruction &insn,
     case amdgpu_gfx90a_op_S_SWAPPC_B64: 
     case amdgpu_gfx940_op_S_SWAPPC_B64: {
         assert(operands.size() == 4);
-        operands.reserve(6);
+        operands.resize(6);
         operands[4] = operands[2];
         operands[5] = operands[3];
```

Comment:
A better solution would be to use `reserve` to make sure the memory is allocated at most once, and then use `push_back` to expand the vector without constructing an operand that is then overwritten.

```cpp
        operands.reserve(6);
        operands.push_back(operands[2]);
        operands.push_back(operands[3]);
```

Reply:
@kupsch I've implemented the suggested fix, and tested that it works.